src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests_ListPatterns.cs

```diff
@@ -0,0 +1,722 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class PatternParsingTests_ListPatterns : ParsingTests
```

Comment:
Please include a test for existing type-pattern using an array type: `_ = o is string[];`
Also variations on `is ..`, such as `is ..var x`, `is ..Type`